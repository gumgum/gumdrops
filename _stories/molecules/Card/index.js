import React from 'react';
import { text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import Card from '../../../components/molecules/Card';
import CardImage from '../../../components/molecules/CardImage';
import CardBlock from '../../../components/molecules/CardBlock';

const cardOptions = {
    white: 'white',
    underlined: 'underlined',
    'underlined-secondary': 'underlined-secondary',
    '': 'No Value'
};

const sizeOptions = {
    xs: 'xs',
    sm: 'sm',
    md: 'md',
    lg: 'lg',
    xl: 'xl',
    '': 'No Value'
};

const component = () => (
    <div>
        <Card
            option={optionalSelect('Option', cardOptions, '')}
            size={optionalSelect('Size', sizeOptions, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}>
            <CardBlock>
                <p>Content in a plain Card.</p>
            </CardBlock>
        </Card>
        <Card
            option={optionalSelect('Option', cardOptions, '')}
            size={optionalSelect('Size', sizeOptions, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}>
            <CardBlock>
                <p>Content in a plain CardBlock.</p>
            </CardBlock>
            <CardBlock option="divide-top">
                <p>Content in a CardBlock with a top divider.</p>
            </CardBlock>
        </Card>
        <Card
            option={optionalSelect('Option', cardOptions, '')}
            size={optionalSelect('Size', sizeOptions, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}>
            <CardImage url="https://c.gumgum.com/ads/com/gumgum/mantii/internal_mobile_inscreen_test/full_canvas/01/mantii_fullscreen.hyperesources/girl@2x.png" />
            <CardBlock option="divide-top">
                <p>Card with a CardImage and a CardBlock.</p>
            </CardBlock>
        </Card>
    </div>
);

export default [readme, component];
