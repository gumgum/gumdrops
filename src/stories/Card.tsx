import React from 'react';
import cx from 'classnames';
import { Sizes } from 'types';

export interface CardProps extends React.HTMLAttributes<Element> {
    option?: 'white' | 'underlined' | 'underlined-secondary';
    size?: Sizes;
}

const baseClass = 'gds-card';

export const Card: React.FC<CardProps> = ({ option, size, className, children, ...otherProps }) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option,
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} {...otherProps}>
            {children}
        </div>
    );
};
