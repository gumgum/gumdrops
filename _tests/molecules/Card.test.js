import renderer from 'react-test-renderer';
import React from 'react';
import Card from '../../components/molecules/Card';

const defaultProps = {
    option: 'underlined-secondary',
    size: 'sm',
    className: 'custom',
    style: { width: 120 }
};

test('Expect <Card> to render properly', () => {
    const tree = renderer
        .create(
            <Card {...defaultProps}>
                <div>Foo</div>
            </Card>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
