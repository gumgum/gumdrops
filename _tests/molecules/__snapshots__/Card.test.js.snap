// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Card> to render properly 1`] = `
<div
  className="gds-card custom gds-card--underlined-secondary gds-card--sm"
  style={
    {
      "width": 120,
    }
  }
>
  <div>
    Foo
  </div>
</div>
`;
